python/prepare_release.py

```diff
@@ -0,0 +1,296 @@
+    UPCOMING_VERSION_REMOTE = 5
+    PULL_REQUEST_CREATED = 6
+
+
+def get_minor_version(version: str) -> str:
+    project_version_details = get_version_details(version)
+    return f'{project_version_details["major"]}.{project_version_details["minor"]}'
+
+
+def update_release(github_token: str, project_name: str, project_version: is_version(str), main_branch: str,
+                   earliest_pr_date: datetime, exec_path: str, is_test: bool = False,
+                   cherry_pick_enabled: bool = False) -> Tuple[str, str, str, ResourceStatus]:
+    multi_extension_sql_path = f"{exec_path}/{MULTI_EXTENSION_SQL}"
+    citus_control_file_path = f"{exec_path}/{CITUS_CONTROL}"
+    multi_extension_out_path = f"{exec_path}/{MULTI_EXTENSION_OUT}"
+    configure_in_path = f"{exec_path}/{CONFIGURE_IN}"
+    config_py_path = f"{exec_path}/{CONFIG_PY}"
+    distributed_dir_path = f"{exec_path}/{DISTRIBUTED_DIR_PATH}"
+
+    resource_status = ResourceStatus.INITIAL
+
+    project_version_details = get_version_details(project_version)
+    default_upcoming_version = get_default_upcoming_version(project_version)
+    upcoming_version = os.getenv("UPCOMING_VERSION", default=default_upcoming_version)
+    upcoming_minor_version = get_minor_version(upcoming_version)
+    devel_version = f"{upcoming_minor_version}devel"
+
+    release_branch_name = f'release-{project_version_details["major"]}.{project_version_details["minor"]}'
+    release_branch_name = f"{release_branch_name}-test" if is_test else release_branch_name
+    upcoming_version_branch = f"master-update-version-{uuid.uuid4()}"
+
+    g = Github(github_token)
+    repository = g.get_repo(f"{REPO_OWNER}/{project_name}")
+    newly_created_sql_file = ""
+
+    # if major release
+    if is_major_release(project_version):
+        print(f"### {project_version} is a major release. Executing Major release flow### ")
+        # create release-X-Y branch
+        print(f"### Preparing {release_branch_name}...### ")
+        # checkout master
+        checkout_branch(main_branch, is_test)
+        # create release branch in release-X.Y format
+        create_branch(release_branch_name)
+        resource_status = ResourceStatus.RELEASE_BRANCH_LOCAL
+        # change version info in configure.in file
+        update_version_in_configure_in(configure_in_path, devel_version)
+
+        # execute "autoconf -f"
+        execute_autoconf_f()
+        # change version info in multi_extension.out
+        update_version_in_multi_extension_out(multi_extension_out_path, devel_version)
+        # commit all changes
+        commit_all_changes_on_branch(project_name, project_version, release_branch_name)
+        # push release branch (No PR creation!!!)
+        if not is_test:
+            push_branch(release_branch_name)
+            resource_status = ResourceStatus.RELEASE_BRANCH_REMOTE
+        print(f"### OK {release_branch_name} prepared.### ")
+        # Increase version number and change upcoming version
+
```

Comment:
Removed comment since method name is descriptive enough. I created two methods
prepare_release_branch_for_major_release
prepare_upcoming_version_branch

Reply:
thanks for doing that